src/poolgirl_internal_directory.erl

```diff
@@ -0,0 +1,66 @@
+%% @private
+%% Poolgirl - A sexy Erlang worker pool factory
+
+-module(poolgirl_internal_directory).
+
+-include_lib("stdlib/include/ms_transform.hrl").
+
+-export([new/0,
+         register/2,
+         find/2]).
+
```

Comment:
This internal directory is what allows for the pool to locate the workers' supervisor (which it needs when recovering from a crash) as well as the top pool supervisor (which it needs to provide when the consumer wishes to stop the pool.)

This hackish workaround is needed because pools can be anonymous (otherwise we could simply name the mentioned supervisors after the pool.)

Anyway: it implies one more ETS table per pool. But this shouldn't be a problem (after all: how many pools can we realistically have?)

The good news is, we no longer have children processes doing blocking calls to their parents, which is generally a footgun waiting to misfire.